Author: Matus Kral <matuskral@me.com>

  regenerate hdmi clk generator on changing VTs as if they have different
  pixel clocks, sound (routed via hdmi) will get messed.

--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -2052,6 +2189,7 @@ static int mxc_hdmi_power_on(struct mxc_dispdrv_handle *disp)
 	dev_dbg(&hdmi->pdev->dev, "%s reg %d cable %d blank %d\n", __func__, hdmi->fb_reg, hdmi->cable_plugin, hdmi->blank);
 
 	mxc_hdmi_phy_init(hdmi);
+	hdmi_clk_regenerator_update_pixel_clock(hdmi->fbi->var.pixclock);
 	return 0;
 }
 
@@ -2336,7 +2475,6 @@ static void mxc_hdmi_setup(struct mxc_hdmi *hdmi, unsigned long event)
 		dev_dbg(&hdmi->pdev->dev, "%s CEA mode\n", __func__);
 
 		/* HDMI Initialization Step E - Configure audio */
-		hdmi_clk_regenerator_update_pixel_clock(hdmi->fbi->var.pixclock);
 		hdmi_enable_audio_clk(hdmi);
 
 		/* HDMI Initialization Step F - Configure AVI InfoFrame */
